sda/cmd/intercept/intercept.go

```diff
@@ -81,6 +81,16 @@ func main() {
 				continue
 			}
 
+			if !isKnownType(msgType) {
+				log.Infof("Unknown message type Routing message (corr-id: %s, routingkey: %s)", delivered.CorrelationId, "catch_all.dead")
+				if err := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, "catch_all.dead", delivered.Body); err != nil {
+					log.Errorf("failed to publish message, reason: (%v)", err)
+				}
+				if err := delivered.Ack(false); err != nil {
+					log.Errorf("failed to ack message for reason: %v", err)
+				}
+			}
+
```

Comment:
While this is a nice solution it will not be triggered.
When the if statement on line 80 is triggered execution will be stopped and the app will fail on timeout.

line 80-82 needs to be replaced with this:
```
if routingKey == "" {
	log.Debugf("msg type: %s", msgType)
	if err := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, "undeliverable", delivered.Body); err != nil {
		log.Errorf("failed to publish message, reason: (%v)", err)
	}
	if err := delivered.Ack(false); err != nil {
		log.Errorf("failed to ack message for reason: %v", err)
	}

	continue
}
```